packages/datatrak-web/src/api/queries/useSurveyResponse.ts

```diff
@@ -2,15 +2,86 @@
  * Tupaia
  *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
  */
+import { useEffect } from 'react';
 import { useQuery } from 'react-query';
-import { DatatrakWebSingleSurveyResponseRequest } from '@tupaia/types';
+import { useFormContext } from 'react-hook-form';
+import { DatatrakWebSingleSurveyResponseRequest, QuestionType } from '@tupaia/types';
 import { get } from '../api';
+import { getAllSurveyComponents, useSurveyForm } from '../../features';
+import { useSurvey } from './useSurvey';
 
-export const useSurveyResponse = (surveyResponseId?: string) => {
+export const useSurveyResponse = (
+  surveyResponseId?: string | null,
+  options?: Record<string, unknown> & { enabled?: boolean },
+) => {
   return useQuery(
     ['surveyResponse', surveyResponseId],
     (): Promise<DatatrakWebSingleSurveyResponseRequest.ResBody> =>
       get(`surveyResponse/${surveyResponseId}`),
-    { enabled: !!surveyResponseId },
+    {
+      enabled: !!surveyResponseId && options?.enabled !== false,
+      ...options,
+    },
   );
 };
+
+/**
+ * Utility hook to fetch survey response data and populate the form with it
+ */
+export const useSurveyResponseWithForm = (surveyResponseId?: string | null) => {
```

Comment:
This is an extension of changes in [this PR](https://github.com/beyondessential/tupaia/pull/5638) as I wanted to avoid too many merge conflicts